__tests__/cmds/custompages.test.js

```diff
@@ -0,0 +1,710 @@
+const nock = require('nock');
+const chalk = require('chalk');
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const frontMatter = require('gray-matter');
+
+const APIError = require('../../src/lib/apiError');
+const getApiNock = require('../get-api-nock');
+
+const CustomPagesCommand = require('../../src/cmds/custompages');
+const SingleCustomPageCommand = require('../../src/cmds/custompages/single');
+
+const custompages = new CustomPagesCommand();
+const customPagesSingle = new SingleCustomPageCommand();
+
+const fixturesBaseDir = '__fixtures__/custompages';
+const fullFixturesDir = `${__dirname}./../${fixturesBaseDir}`;
+const key = 'API_KEY';
+
+function hashFileContents(contents) {
+  return crypto.createHash('sha1').update(contents).digest('hex');
+}
+
+describe('rdme custompages', () => {
+  beforeAll(() => nock.disableNetConnect());
+
+  afterAll(() => nock.cleanAll());
+
+  it('should error if no api key provided', () => {
+    return expect(custompages.run({})).rejects.toStrictEqual(
+      new Error('No project API key provided. Please use `--key`.')
+    );
+  });
+
+  it('should error if no folder provided', () => {
+    return expect(custompages.run({ key })).rejects.toStrictEqual(
+      new Error('No folder provided. Usage `rdme custompages <folder> [options]`.')
+    );
+  });
+
+  it('should error if the argument is not a folder', () => {
+    return expect(custompages.run({ key, version: '1.0.0', folder: 'not-a-folder' })).rejects.toThrow(
+      "ENOENT: no such file or directory, scandir 'not-a-folder'"
+    );
+  });
+
+  it('should error if the folder contains no markdown nor HTML files', () => {
+    return expect(custompages.run({ key, folder: '.github/workflows' })).rejects.toStrictEqual(
+      new Error('We were unable to locate Markdown or HTML files in .github/workflows.')
+    );
+  });
+
+  describe('existing custompages', () => {
+    let simpleDoc;
+    let anotherDoc;
+
+    beforeEach(() => {
+      let fileContents = fs.readFileSync(path.join(fullFixturesDir, '/existing-docs/simple-doc.md'));
+      simpleDoc = {
+        slug: 'simple-doc',
+        doc: frontMatter(fileContents),
+        hash: hashFileContents(fileContents),
+      };
```

Comment:
Super nitpicky but might be cleaner to do these as a promise?

```suggestion
simpleDoc = await fsPromises.readFile(path.join(fullFixturesDir, '/existing-docs/simple-doc.md')).then(contents => ({
  slug: 'simple-doc',
  doc: frontMatter(contents),
  hash: hashFileContents(contents),
});
```

Reply:
feels like two steps forward two steps back so gonna hold off on this!

Reply:
lol